pkg/api/types.go

```diff
@@ -1309,3 +1317,27 @@ type ResourceQuotaList struct {
 	// Items is a list of ResourceQuota objects
 	Items []ResourceQuota `json:"items"`
 }
+
+// Secret holds secret data of a certain type
+type Secret struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	Data map[string][]byte `json:"data,omitempty"`
```

Comment:
@thockin go for it, I was not proposing that you address that in #4653

On Mon, Feb 23, 2015 at 2:25 PM, Tim Hockin <notifications@github.com>
wrote:

> In pkg/api/types.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/4514#discussion_r25192983>
> :
>
> > @@ -1309,3 +1317,27 @@ type ResourceQuotaList struct {
> >  	// Items is a list of ResourceQuota objects
> >  	Items []ResourceQuota `json:"items"`
> >  }
> > +
> > +// Secret holds secret data of a certain type
> > +type Secret struct {
> > +	TypeMeta   `json:",inline"`
> > +	ObjectMeta `json:"metadata,omitempty"`
> > +
> > +	Data map[string][]byte `json:"data,omitempty"`
>
> I'm not sure that the "always base64" crumminess is worse than the
> "sometimes base64" crumminess :)
>
> Anyway, I'd like to get #4653
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/4653> in before
> any significant retool, please. I just want it off the balance sheet, and
> it fixes many of the concerns here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/4514/files#r25192983>
> .
>